packages/eslint-plugin-gestalt/src/no-workflow-status-icon.js

```diff
@@ -0,0 +1,114 @@
+/**
+ * @fileoverview Error on use a <Icon icon="workflow-status-[...]" size={16} />, and suggest use
+ * <Status /> component rather than <Icon />
+ */
+
+// @flow strict
+import { type ESLintRule } from './helpers/eslintFlowTypes.js';
+import { hasImport, isGestaltComponent, hasAttributes } from './helpers/eslintASTHelpers.js';
+
+const disallowedMatch = [
+  { icon: 'workflow-status-in-progress', color: 'success' },
+  { icon: 'workflow-status-ok', color: 'success' },
+  { icon: 'workflow-status-problem', color: 'error' },
+  { icon: 'workflow-status-warning', color: 'warning' },
+];
+
+export const errorMessage =
+  'Icons where `workflow-status-[...]` and matching color should be replaced for the Status Component. https://gestalt.pinterest.systems/status';
```

Comment:
I'm not the best English proofreader, but I think  "Icons where `workflow-status-[...]` and matching color should be replaced for the Status Component" is not 100% correct in terms of comprehension.

Again,  I'd simplify to 

This instance of  workflow-status-[...] Icon matches a type of Status. Please, replace with Status.